enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,253 @@
+SAR requests to the spoke cluster, but this should be fine b/c the work-agent 
+is working asynchronously. 
+
+Note that the authorization via subject access review api doesn't require the
+service account to be actually present in the managed cluster, so in the cases
+where you have too many executor subject to manage, the number of service 
+accounts resources in the managed cluster is bounded.
+
+A sample of the requesting `SubjectAccessReview` payload is:
+
+```yaml
+apiVersion: authentication.open-cluster-management.io/v1alpha1
+kind: SubjectAccessReview
+spec:
+  # requester is the referenced service-account in executor subject
+  user: "system:serviceaccount:<sa namespace>:<sa name>"
+  groups: ["system:serviceaccounts", "system:authenticated"]
+  # requesting resource attributes
+  resourceAttributes:
+    # verb
+    verb: <the intended verb, can be create/update/patch/delete>
+    # the GVR tuple
+    group: <api group>
+    version: <api version>
+    resource: <api resource name>
+    # the object identifier
+    name: <resource name>
+    namespace: <resource namespace if applicable>
+```
+
+### Architecture
+
+The overall architecture is shown in the picture below:
+
+![Architecture](./arch.png "Architecture")
+
+#### Workflow
+
+The following is what will happen when a `ManifestWork` with non-nil executor
+subject specified is applied to the hub cluster:
+
+0. Create a "ManifestWork" with non-nil executor subject
+1. The work-agent observes the new resource via watch event
+2. The work-agent builds a SubjectAccessReview request which:
+   - The target subject is the executor subject from `ManifestWork`
+   - The target attribute should contain the "*" verbs for all the inlined
+     objects in the spec of `ManifestWork`. Note that the "resourceNames"
+     should also be specified.
+3. The work-agent writes the objects to the managed cluster if SAR passes.
+   3.1 Otherwise set the failure condition to the `ManifestWork`.
+4. The work-agent reports the results back to the hub cluster.
+
+### Proposing changes to the work agent
+
+
+### Backward Compatibility
+
+Considering the data-plane backward compatibility, those existing legacy 
+`ManifestWork` resources will be considered as nil executor subject, so
+the work agent keeps using the original service account.
+
+
+### Authorizing hub-user to executor subjects
+
+#### Admission control via `SubjectAccessReview`
+
+A validating webhook admission controller that checks whether the requesting
+hub users has sufficient permission to manipulate the work object:
+
+```yaml
+apiVersion: authentication.open-cluster-management.io/v1alpha1
+kind: SubjectAccessReview
+spec:
+  user: <requester username>
+  groups: <requester groups>
+  resourceAttributes:
+    # virtual verb
+    verb: execute-as
+    # the GVR tuple of the requesting resources
+    group: work.open-cluster-management.io
+    version: v1
+    resource: manifestworks
+    namespace: <cluster namespace>
```

Comment:
I think I understand the reasoning for this. This is chosen instead of the service account namespace because this allows for delegation inside of a single managed cluster.  this reasoning is worth commenting here.

Reply:
yes the managed cluster admin should also be able to grant/revoke accessibility of an executor-subject from the hub users. i added that in the new commit.